idstools/unified2.py

```diff
@@ -813,23 +854,16 @@ def next(self):
 
         # Now try to get a record.  If we can't see if there is a new
         # file and try again.
-        try:
-            record = self.reader.next()
-        except EOFError:
-            return
-        if record:
-            return record
-        else:
-            while True:
-                if self.open_next():
-                    try:
-                        record = self.reader.next()
-                    except EOFError:
-                        return
-                    if record:
-                        return record
-                else:
-                    return None
+        while True:
+            try:
+                record = self.reader.next()
+            except EOFError:
+                return
+            if record:
+                return record
+            else:
+                if not self.open_next():
+                    return
```

Comment:
Is this required for the new record type? If not, please break it out into its own commit please.